qiskit_ibm_runtime/sampler.py

```diff
@@ -172,6 +174,10 @@ def __init__(
         _options.transpilation.skip_transpilation = (  # type: ignore[union-attr]
             skip_transpilation
         )
+        if _options.resilience_level and not _options.resilience_level in [0, 1]:
+            raise ValueError(
+                "resilience level can only take the values [0, 1] in Sampler"
+            )
```

Comment:
Should be the same in both. The question is whether I should check the options in `__init__` or if checking in `run` is sufficient (for both `Sampler` and `Estimator`. I would think checking in `run` is sufficient, but I think some of the existing tests would fail (of course, I can change the tests).